### ISSUE TITLE
[Bug]: Error when configuring Recycle Coach

### ISSUE THREAD

**Issue by cdn4lf** [mampfes/hacs_waste_collection_schedule#1596]
### I Have A Problem With:

A specific source

### What's Your Problem

Receiving error when configuring recycle coach for Vaughan Ontarion (unlisted city, but supported by recycle coach)

### Source (if relevant)

recycle coach

### Logs

```Shell
Logger: waste_collection_schedule.source_shell
Source: custom_components/waste_collection_schedule/waste_collection_schedule/source_shell.py:136
Integration: waste_collection_schedule (documentation)
First occurred: 3:10:21 PM (1 occurrences)
Last logged: 3:10:21 PM

fetch failed for source Recycle Coach: Traceback (most recent call last): File "/config/custom_components/waste_collection_schedule/waste_collection_schedule/source_shell.py", line 134, in fetch entries = self._source.fetch() ^^^^^^^^^^^^^^^^^^^^ File "/config/custom_components/waste_collection_schedule/waste_collection_schedule/source/recyclecoach_com.py", line 159, in fetch self._lookup_zones() File "/config/custom_components/waste_collection_schedule/waste_collection_schedule/source/recyclecoach_com.py", line 135, in _lookup_zones for zone_res in zone_data["results"]: ~~~~~~~~~^^^^^^^^^^^ KeyError: 'results'
```

### Relevant Configuration

```YAML
waste_collection_schedule:
  sources:
  - name: recyclecoach_com 
    calendar_title: Waste Collection Schedule
    args:
      street: <<street address in lowercase>>
      city: <<city in lowercase>>
      state: <<province in lowercase>>
```

### Checklist Source Error

- [X] Use the example parameters for your source (often available in the documentation) (don't forget to restart Home Assistant after changing the configuration)
- [X] Checked that the website of your service provider is still working
- [X] Tested my attributes on the service provider website (if possible)
- [X] I have tested with the latest version of the integration (master) (for HACS in the 3 dot menu of the integration click on "Redownload" and choose master as version)

### Checklist Sensor Error

- [ ] Checked in the Home Assistant Calendar tab if the event names match the types names (if types argument is used)

### Required

- [X] I have searched past (closed AND opened) issues to see if this bug has already been reported, and it hasn't been.
- [X] I understand that people give their precious time for free, and thus I've done my very best to make this problem as easy as possible to investigate.

**Comment by 5ila5** [comment]
I can not reproduce this issue as all our test cases pass and return collection dates.
Our test cases:
```json
{
    "Default": {"street": "2242 grinstead drive", "city": "louisville", "state": "KY"},
    "Problematic City Lookup": {
        "street": "2202 E Florence Dr",
        "city": "Tucson",
        "state": "AZ",
        "district_id": "TUC",
        "project_id": "532",
    },
    "olympia": {
        "street": "1003 Lybarger St NE",
        "city": "Olympia",
        "state": "Washington",
    },
    "newark": {"street": "24 Townsend Rd", "city": "Newark", "state": "Delaware"},
    "albuquerque": {
        "street": "1505 Silver Ave SE",
        "city": "Albuquerque",
        "state": "New Mexico",
    },
    "london ontario": {
        "street": "1065 Sunningdale Rd E",
        "city": "London",
        "state": "Ontario",
    },
    "london ontario with districtID": {
        "street": "1065 Sunningdale Rd E",
        "city": "London",
        "state": "Ontario",
        "project_id": "528",
        "district_id": "CityofLondon",
        "zone_id": "zone-z547",
    },
    "aurora ontario": {
        "street": "123 Cranberry Lane",
        "city": "Aurora",
        "state": "Ontario",
    },
}
```
could you try one of them and look if you st

**Comment by cdn4lf** [comment]
I used the Aurora test case and it runs without an issue.  When I use the same format for Vaughan, which I know is using recycle coach, it fails with the error above.

Is there a way to get you the address for use in testing and/or a way for me to validate the address better?  I can't find API docs on how to query an address.

**Comment by cdn4lf** [comment]
Adding.  When I add a district and project, I'm getting the following error.

`Logger: homeassistant.setup
Source: setup.py:332
First occurred: 5:26:44 PM (1 occurrences)
Last logged: 5:26:44 PM

Error during setup of component waste_collection_schedule
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/setup.py", line 332, in _async_setup_component
    result = await task
             ^^^^^^^^^^
  File "/config/custom_components/waste_collection_schedule/__init__.py", line 109, in async_setup
    api.add_source_shell(
  File "/config/custom_components/waste_collection_schedule/__init__.py", line 202, in add_source_shell
    SourceShell.create(
  File "/config/custom_components/waste_collection_schedule/waste_collection_schedule/source_shell.py", line 196, in create
    source = source_module.Source(**source_args)  # type: ignore
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/waste_collection_schedule/waste_collection_schedule/source/recyclecoach_com.py", line 91, in __init__
    self.project_id = self._format_key(project_id) if project_id else None
                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/waste_collection_schedule/waste_collection_schedule/source/recyclecoach_com.py", line 99, in _format_key
    return param.upper().strip()
           ^^^^^^^^^^^
AttributeError: 'int' object has no attribute 'upper'
`

**Comment by 5ila5** [comment]
You can test addresses here: https://app.my-waste.mobi/widget/576-Vaughan/home.php
It looks like Vaughan uses an other method to get the zone id (using coordinates) and a different zone format (z before numbers) I added support for the geo API in #1601 so that your configuration with providing street, city and state should work.

tested with:
```json
{
        "street": "Main St",
        "city": "Vaughan",
        "state": "Ontario",
}
```

> Adding. When I add a district and project, I'm getting the following error.

the sources expected a string but you provided an int you could have fixed this by adding " around the id. I changed this as well in #1601 so that integers should be fine

This fix will be part of 1.45.0 or is available now as master (git) version (3-dot menu -> redownload -> change version to master -> fully restart HA (this will increase update cycle by a lot)

**Comment by cdn4lf** [comment]
This works perfectly with the upcoming release.  

Not sure if you want to add Vaughan Ontario as a confirmed source but it's working perfectly.

Thank you for this, truely fantastic
